File: C++/23. Merge k Sorted Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeKLists(vector<ListNode*>& lists) {
        vector<int> vec;
        for (int i = 0; i < lists.size(); i++) {
            while(lists[i] != NULL){
                vec.push_back(lists[i]->val);
                lists[i] = lists[i]->next;
            }
        }
        sort(vec.begin(),vec.end());
        ListNode* temp = new ListNode();
        ListNode* ans = temp;
        for (int i: vec){
            temp->next = new ListNode(i);
            temp=temp->next;
        }
        return ans->next;
    }
};

File: JavaScript/23. Merge k Sorted Lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode[]} lists
 * @return {ListNode}
 */
var mergeKLists = function(lists) {
    var arr = new Array();
    for (var i = 0; i < lists.length; i++) {
        while(lists[i] != null){
            arr.push(lists[i].val);
            lists[i] = lists[i].next;
        }
    }
    arr.sort((a,b) => a - b)
    var temp = new ListNode();
    var ans = temp;
    for (var i of arr){
        temp.next = new ListNode(i);
        temp=temp.next;
    }
    return ans.next;
};
